packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -190,6 +191,52 @@ export const getParticipants = async (req: Request, res: Response) => {
   }
 };
 
+export const updateParticipant = async (req: Request, res: Response) => {
+  try {
+    // valid giveaway
+    const giveaway = await Giveaway.findById(req.params.id);
+    if (!giveaway) return res.status(404).json({ error: 'Giveaway not found' });
+
+    // valid participant
+    const participant = giveaway.participants.find(
+      (participant) => participant.id === req.params.participantId
+    );
+    if (!participant)
+      return res.status(404).json({ error: 'Participant not found' });
+    if (participant.state !== ParticipantState.PENDING)
+      return res.status(400).json({ error: 'Participant state already set' });
+
+    // valid state
+    const { state } = req.body;
+    if (!Object.values(ParticipantState).includes(state))
+      return res
+        .status(400)
+        .json({ error: `${state} is not a valid participation state` });
+
+    // save participant
+    participant.state = state;
+    await giveaway.save();
+
+    // add to smart contract
+    try {
+      if (state === ParticipantState.CONFIRMED) {
+        const participantHash = encrypt(participant.id);
+        await giveawaysContract.methods
+          .addParticipant(objectIdToBytes24(giveaway._id), participantHash)
+          .send({ from: process.env.OWNER_ACCOUNT_ADDRESS, gas: '1000000' });
+      }
+    } catch (error) {
+      participant.state = ParticipantState.PENDING;
+      await giveaway.save();
+      throw error;
```

Comment:
Instead of having the try catch here we can just save the participant at the end, and only save to the database if everything goes right, without having to rollback